src/logger.js

```diff
@@ -1,29 +1,68 @@
 import { TracerGlobals } from './globals';
-
+import { isStoreLogs } from './utils';
 const LOG_PREFIX = '#LUMIGO#';
 const WARN_CLIENT_PREFIX = 'Lumigo Warning';
 
+const MAX_DUPLICATE_LOGS = 50;
+
+export const LogStore = (() => {
+  let logSet = new Set([]);
+  let duplicateLogsCount = 0;
+
+  const addLog = (type, message, object) => {
+    const logObj = JSON.stringify({ type, message, object });
+    if (!logSet.has(logObj)) {
+      logSet.add(logObj);
+    } else {
+      duplicateLogsCount++;
+    }
+    isEmergencyMode() && printLogs();
+  };
+
+  const printLogs = () => {
+    logSet.forEach(logObj => {
+      const { message, obj } = JSON.parse(logObj);
+      forceLog('FATAL', message, obj);
+    });
+    logSet.clear();
+  };
+
+  const isEmergencyMode = () => duplicateLogsCount >= MAX_DUPLICATE_LOGS;
```

Comment:
Do we identify an infinite loop by checking if there are duplicates of logs?
There could be an infinite loop without logs and there could be logs without an infinite loop.
Are we ok with this?

Reply:
Yes, if you have other ideas on how to figure we in an infinite loop you can suggest :)